### PR TITLE
refactor(backend): refactor code by standard

### LINKED ISSUE #47: Refactor backend
## Description
Refactor backend to the standard

- prettier
- eslint
- typescript
- validations
- connection to multiple dbs

## DoD

- [ ] Backend is refactored fully

### PR DESCRIPTION
**Pull request by chenyair**
closes #47

### CODE REVIEW

File: backend/src/db.ts
Comment (Shaked-abdu) [inline backend/src/db.ts:1]: awesome !

**Review by Shaked-abdu — APPROVED** [review]

**Comment by Shaked-abdu** [comment]
Maybe delete the README file or update him

File: backend/package.json [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
+    "@types/bcrypt": "^5.0.2",
+    "@types/cors": "^2.8.17",
+    "@types/jsonwebtoken": "^9.0.6",
+    "@types/node": "^20.14.8",
+    "@types/swagger-jsdoc": "^6.0.4",
+    "@types/swagger-ui-express": "^4.1.6",
+    "@types/uuid": "^10.0.0",
+    "@typescript-eslint/eslint-plugin": "^7.13.1",
+    "@typescript-eslint/parser": "^7.13.1",
+    "dev": "TZ='UTC' nodemon src/index.ts",
```
Comment (liadyacobi) [inline backend/package.json]: what is this dependency?

File: backend/src/controllers/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+const refresh = async (req: Request, res: Response<IUserTokens | { message: string }>) => {
+  const authHeader = req.headers['authorization']
+  const token = authHeader && authHeader.split(' ')[1]
+
+  if (!token) {
+    res.status(401).json({ message: 'Token not provided' })
+    return
+  }
+
+  jwt.verify(token, config.refreshTokenSecret, async (err, clientInfo) => {
```
Comment (liadyacobi) [inline backend/src/controllers/auth.controller.ts:94]: does this support async await or do we have to use callbacks?

File: backend/src/app.ts [hunk 3]
```diff
@@ -1,28 +1,47 @@
...
-      .then(() => {
-        resolve(app);
-      })
-      .catch((err) => {
-        reject(err);
-      });
-  });
-};
+  await connectDatalake()
+  await connectMetadata()
```
Comment (liadyacobi) [inline backend/src/app.ts]: If you want you can initiate both async using `Promise.all` and calling on both connections

File: backend/.prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "tabWidth": 2,
+    "semi": false,
+    "singleQuote": true,
+    "trailingComma": "none",
```
Comment (liadyacobi) [inline backend/.prettierrc.json:5]: why no `;` at end of line?

**Review by liadyacobi — COMMENTED** [review]
Looks great

File: backend/.prettierrc.json (hunk 4 above)
Comment (liadyacobi) [inline backend/.prettierrc.json]: change to comma es5
```suggestion
    "trailingComma": "es5",
```

File: backend/.prettierrc.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "tabWidth": 2,
+    "semi": false,
```
Comment (liadyacobi) [inline backend/.prettierrc.json]:
```suggestion
    "semi": true,
```

**Pull request merged**